### PR TITLE
Check anr rpc version

### PR DESCRIPTION
**Pull request by felipemadero**
Closes https://github.com/ava-labs/avalanche-cli/issues/586

### CODE REVIEW

File: pkg/binutils/processes.go [hunk 1]
```diff
@@ -64,6 +64,22 @@ func NewGRPCClient() (client.Client, error) {
...
+		rpcVersion, err := client.RPCVersion(ctx)
+		if err != nil {
+			return nil, err
+		}
+		// obtained using server API
+		serverVersion := rpcVersion.Version
+		// obtained from ANR source code
+		clientVersion := server.RPCVersion
+		if serverVersion != clientVersion {
+			return nil, fmt.Errorf("trying to connect to a backend controller that uses a different RPC version (%d) than the CLI client (%d). Use network clean to stop the controller and then restart the operation",
```
Comment (dasconnor) [inline pkg/binutils/processes.go:78]: hmm, do we want a way to do this that does not require clearing your deployed subnets? Maybe like a `network clean --server-only` flag that preserves your snapshot?

Comment (felipemadero) [reply pkg/binutils/processes.go:78]: mmm do you prefer  `network clean --server-only` (need to add snapshot preservation to clean) or a `network stop --shutdown-server` (will add server shutdown to stop)

Comment (dasconnor) [reply pkg/binutils/processes.go:78]: `network stop --shutdown-server` is better

File: cmd/networkcmd/stop.go [hunk 2]
```diff
@@ -59,5 +60,12 @@ func StopNetwork(*cobra.Command, []string) error {
 		return fmt.Errorf("failed to stop network with a snapshot: %w", err)
 	}
 	ux.Logger.PrintToUser("Network stopped successfully.")
+
+	if err := binutils.KillgRPCServerProcess(app); err != nil {
+		app.Log.Warn("failed killing server process", zap.Error(err))
+	} else {
+		ux.Logger.PrintToUser("Server process terminated.")
```
Comment (dasconnor) [inline cmd/networkcmd/stop.go:67]: maybe reword to `Server shutdown gracefully`

**Review by dasconnor — APPROVED** [review]
one small suggestion, then looks good
